src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -69,15 +72,21 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
 
             try
             {
-                HttpResponse response = await _httpManager.SendGetAsync(_ImdsUri, Headers, logger).ConfigureAwait(false);
+                HttpResponse response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+
+                if (response.StatusCode != HttpStatusCode.BadRequest)
```

Comment:
Please add a comment about this. When would this be a "bad request" ?

Reply:
Added